openedx/MITx-edx-integration-tutor.md

```diff
@@ -0,0 +1,199 @@
+
+    tutor dev do createuser --staff mit_Application_serviceworker service@mitx.odl.local
+
+Supply a password (this one doesn't matter for a local deployment, you won't ever actually use the account).
+
+### For MITxOnline Only
+For best results, create two new courses within edX. The MITxOnline `configure_instance` command expects a couple of courses to exist in edX (because they come with the devstack package):
+
+| Course ID                       | Course Title         |
+|---------------------------------|----------------------|
+| course-v1:edX+DemoX+Demo_Course | Demonstration Course |
+| course-v1:edX+E2E-101+course    | E2E Test Course      |
+
+If you have a devstack instance handy, you can export these and import them into Tutor. Otherwise, just create them and make sure to set dates for the courses (they default to 2030 otherwise).
+
+### Configure Open edX user and token for use with MITx Application management commands
+
+- In Open edX, under `/admin/oauth2_provider/accesstoken/` add access token with that newly created staff user.
+
+### MIT Application Setup
+
+To set up MIT Application:
+
+1. Get the gateway IP for the `EDX_APP`
+   1. Linux users: The gateway IP of the docker-compose networking setup for edx LMS `docker network inspect tutor_dev_default | grep Gateway`.
+   2. OSX users: Use `host.docker.internal`
+
+2. Set up your `.env` file. These settings need particular attention:
+
+   - `OPENEDX_IP`: set to the gateway IP from the first step.
+   - `OPENEDX_API_BASE_URL`: set to `http://<EDX_HOSTNAME>:<PORT>`
+   - `OPENEDX_SERVICE_WORKER_USERNAME`: set to `mit_Application_serviceworker` (unless you changed this)
+   - `OPENEDX_SERVICE_WORKER_API_TOKEN`: set to the token you just generated
+
+### Run the configure_instance command
+
+    docker-compose run --rm web ./manage.py configure_instance linux --gateway <ip> --tutor-dev
+
+  Where `<ip>` is the IP from the first step.(On macOS, specify macos instead of linux. You can also skip --gateway.) You will need to supply passwords for the MITx Application superuser and test learner accounts. **Make a note of the client ID and secret that it will print out at the end.**
+
+
+## Tutor Setup, Part Two
+
+Note that some of these steps require editing the main configuration files for the production instance (which is also used for a local deployment). Most of the settings that need to be adjusted to get integration working are overridden by the default Tutor configuration, so you can't update them by setting `config.yml`. If you're using the development Tutor build, you'll likely need to edit `development.py` rather than `production.py` as necessary.
+
+These steps will also disable the AuthN SSO MFE, so from here on you'll get normal edX authentication screens (if you're not being bounced to MITx Application).
+
+
+1. Get the gateway IP of the `mitxApplication_default` Docker network:
+
+       docker network inspect mitxpro_default | grep Gateway
```

Comment:
Nit: this uses `mitxpro` where the convention seems to be `mitxApplication`.

Reply:
Actually its an example, I will update it as ~~Docker network:~~ `Docker network example:`